savant/deepstream/pipeline.py

```diff
@@ -376,9 +405,11 @@ def _add_input_converter(
             source_info.before_muxer.append(video_converter)
 
         source_info.before_muxer.append(nv_video_converter)
+        self._check_pipeline_is_running()
         # TODO: send EOS to video_converter on unlink if source didn't
         assert new_pad.link(video_converter_sink) == Gst.PadLinkReturn.OK
 
+        self._check_pipeline_is_running()
```

Comment:
If I understand correctly, the method `_check_pipeline_is_running()` seems to be called repeatedly to account for possible changes to the `self._is_running` flag from parallel threads.

Isn't it better to use some synchronization mechanism to ensure immutability for this flag during the `_add_input_converter()` method (and other same casese) call?

Reply:
We can add a synchronization on a case when user stops the module. But when the module fails with an error synchronization would be difficult to add.